Description: Use x-terminal-emulator in preference to xterm
Author: James Cowgill <jcowgill@debian.org>
Author: Alessio Treglia <alessio@debian.org>
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/LiteEditor/manager.cpp
+++ b/LiteEditor/manager.cpp
@@ -1959,7 +1959,7 @@ void Manager::DbgStart(long attachPid)
     wxString terminal;
     wxArrayString tokens;
     wxArrayString configuredTerminal;
-    terminal = wxT("xterm");
+    terminal = wxT("x-terminal-emulator");
     if(!EditorConfigST::Get()->GetOptions()->GetProgramConsoleCommand().IsEmpty()) {
         tokens =
             wxStringTokenize(EditorConfigST::Get()->GetOptions()->GetProgramConsoleCommand(), wxT(" "), wxTOKEN_STRTOK);
--- a/LiteEditor/app.cpp
+++ b/LiteEditor/app.cpp
@@ -201,7 +201,9 @@ static void WaitForDebugger(int signo)
         char command[256];
         memset(command, 0, sizeof(command));
 
-        if(ExeLocator::Locate(wxT("gnome-terminal"), where)) {
+        if(ExeLocator::Locate(wxT("x-terminal-emulator"), where)) {
+            sprintf(command, "x-terminal-emulator -T 'gdb' -e 'gdb -p %d'", getpid());
+        } else if(ExeLocator::Locate(wxT("gnome-terminal"), where)) {
             sprintf(command, "gnome-terminal -t 'gdb' -e 'gdb -p %d'", getpid());
         } else if(ExeLocator::Locate(wxT("konsole"), where)) {
             sprintf(command, "konsole -T 'gdb' -e 'gdb -p %d'", getpid());
